### PR TITLE
feat(migration): Add comment actions. 

### PR DESCRIPTION
**Pull request by GeorginaFraire**
#### What does this PR do?
- Implement all the actions related with the comments: 
1 - Create
2 - Update
3 - Delete
4 - Vote (up and down)
5 - Sort 

> Note: Approve a comment is not available in this PR. 

#### How should this be manually tested?
1. Login
2. Select any question and go to its detail
3. Create a comment (user and anonymous)
4. Modify the comment created.
5. Delete the comment created.
6. Create more comments
7. Votes for different comments (up and down)
8. Sort the comments.

#### Screenshots
[Create]

https://user-images.githubusercontent.com/97203617/199144765-1cfec425-d708-4566-a12e-4fb2f99ae12b.mov

https://user-images.githubusercontent.com/97203617/199144962-f0415d49-887d-40cc-9fb6-00a37cb5f2fb.mov

[Update]

https://user-images.githubusercontent.com/97203617/199144647-20e4a920-4cf3-4ad5-bafb-06a5a9e8af77.mov

[Delete]

https://user-images.githubusercontent.com/97203617/199144670-a0412e8f-d2ae-4cc0-8c84-8935aa52579f.mov

[Vote & Sort]

https://user-images.githubusercontent.com/97203617/199144689-780ead8b-3701-444d-82c6-72f2f383fff0.mov

### CODE REVIEW

**Comment by GeorginaFraire** [comment]
> @GeorginaFraire Great job 🎉 Just have some observations from testing locally.
> 
> 1. When typing a comment, markdown tooltip is not shown as before above "Show preview":
> 
> Remix: image
> 
> Before: image
> 
> 2. Every time a user likes a comment, a toast message is shown which could be a little excesive. This could be fixed by removing success message in the response of like comment controller.
>    image
> 3. When url has a set order type (Example http://localhost:3000/questions/193?order=activityTime) it orders correctly, but "Order By" dropdown shows default option instead (fixed by re-selecting in dropdown).
> 
> Screen Shot 2022-11-01 at 16 04 39
> 
> Should we add these as new tickets?

I don't think so, I will fix the issues in this PR.

**Comment by GeorginaFraire** [comment]
@albertcastaned Can you review it again, please?

https://user-images.githubusercontent.com/97203617/199369041-3f134381-bd25-4c75-ae10-27707f7ecc23.mov

**Review by albertcastaned — APPROVED** [review]

**Comment by GeorginaFraire** [comment]
> image Getting this error when refreshing the page :(. I think we can not use `window` because in Remix we do Server-Side Rendering and this does not exist from "server". We could just set it to a static value or look if there is an alternative

Done!
Fixed by adding a static value.

File: app/controllers/commentVotes/voteComment.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ export const upsertCommentVote = async (query) => {
     });
 
     return {
-      success: 'Vote updated successfully!',
+      success: true,
```
[the remark this answers is not in the document]
Comment (GeorginaFraire) [reply app/controllers/commentVotes/voteComment.js]: Done!

File: app/components/QuestionComment/QuestionComment.jsx [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+    const [searchParams] = useSearchParams();
+    const isAnswer = props.hasCommentAsAnswer;
+  
+    const {
+      state: showCommentOptionsState,
+      setState: setShowCommentOptions,
+      wrapperRef,
+    } = useClickOutside(isEditing || isDeleting);
+    
+    async function voteComment (comment_id, value){
```
Comment (Hoosep) [inline app/components/QuestionComment/QuestionComment.jsx:49]: async is necessary? I don't see any await inside this function.

**Review by Hoosep — APPROVED** [review]

File: app/components/QuestionComment/QuestionComment.jsx (hunk 2 above)
Comment (albertcastaned) [reply app/components/QuestionComment/QuestionComment.jsx:49]: It was before migration to Remix but think its not necesary anymore, good spot

Comment (GeorginaFraire) [reply app/components/QuestionComment/QuestionComment.jsx]: Done! Thanks

**Review by albertcastaned — APPROVED** [review]
Great work 💯

**Pull request merged**